src/main/scala/subsystem/HasTiles.scala

```diff
@@ -22,6 +25,96 @@ trait HasTiles { this: BaseSubsystem =>
   def hartIdList: Seq[Int] = tileParams.map(_.hartId)
   def localIntCounts: Seq[Int] = tileParams.map(_.core.nLocalInterrupts)
   def sharedMemoryTLEdge = sbus.busView
+
+  protected def augmentedTileParameters(tp: TileParams): Parameters = p.alterPartial {
+    // For legacy reasons, it is convenient to store some state
+    // in the global Parameters about the specific tile being built now
+    case TileKey => tp
+    case SharedMemoryTLEdge => sharedMemoryTLEdge
+  }
+
+  protected def connectMasterPortsToSBus(tile: BaseTile, crossing: RocketCrossingParams) {
```

Comment:
Should `RocketCrossingParams` become `TileCrossingParams`?

Reply:
Yes, I think so. I didn’t want to force that change as part of this refactor.